atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraReloadableKvsConfig.java

```diff
@@ -115,4 +115,17 @@ public int concurrentGetRangesThreadPoolSize() {
                 .map(_runtime -> poolSize() * servers().numberOfThriftHosts())
                 .orElseGet(config::concurrentGetRangesThreadPoolSize);
     }
+
+    @Override
+    public int defaultGetRangesConcurrency() {
+        if (config.servers().numberOfThriftHosts() > 0) {
+            return config.defaultGetRangesConcurrency();
+        }
+
+        // Use the initial number of thrift hosts as a best guess
+        return runtimeConfigSupplier
+                .get()
+                .map(_runtime -> poolSize() * servers().numberOfThriftHosts())
```

Comment:
I think this needs to be
```suggestion
                .map(_runtime -> Math.min(8, concurrentGetRangesThreadPoolSize() / 2))
```